frameworklauncher/src/main/java/com/microsoft/frameworklauncher/service/StatusManager.java

```diff
@@ -482,12 +533,15 @@ public synchronized void transitionFrameworkState(
     frameworkStatus.setFrameworkState(dstState);
 
     // Update ZK Status
-    zkStore.setFrameworkStatus(frameworkName, frameworkStatus);
-    LOGGER.logInfo("Transitioned Framework [%s] from [%s] to [%s]", frameworkName, srcState, dstState);
+    if (!event.getSkipToPersist()) {
+      zkStore.setFrameworkStatus(frameworkName, frameworkStatus);
```

Comment:
>zkStore.setFrameworkStatus(frameworkName, frameworkStatus); [](start = 6, length = 59)

When will the stopped framework to be persisted?

Reply:
At line 338

---
In reply to: [172149878](https://github.com/Microsoft/pai/pull/346#discussion_r172149878) [](ancestors = 172149878)